### PR TITLE
[core] Added more logs around accept errors

### LINKED ISSUE #1880: [maint] Improve error information displayed at the accept activities
There are no messages displayed in the logs during the call of `srt_accept`. It would be nice to have this information to help developers determine problems, including rejected connections.

### PR DESCRIPTION
**Pull request by ethouris**
Fixes #1880 

Added more logs around errors from `srt_accept`.

### CODE REVIEW

File: srtcore/api.cpp [hunk 1]
```diff
@@ -1065,16 +1065,25 @@ SRTSOCKET CUDTUnited::accept_bond(const SRTSOCKET listeners [], int lsize, int64
...
+   {
+      LOGC(cnlog.Error, log << "srt_accept: incorrect address parameters specified");
       throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+   }
 
    CUDTSocket* ls = locateSocket(listen);
 
    if (ls == NULL)
+   {
+      LOGC(cnlog.Error, log << "srt_accept: invalid listener socket value: " << listen);
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1077]:
```suggestion
      LOGC(cnlog.Error, log << "srt_accept: invalid listener socket ID value: " << listen);
```

File: srtcore/api.cpp [hunk 2]
```diff
@@ -1065,16 +1065,25 @@ SRTSOCKET CUDTUnited::accept_bond(const SRTSOCKET listeners [], int lsize, int64
 SRTSOCKET CUDTUnited::accept(const SRTSOCKET listen, sockaddr* pw_addr, int* pw_addrlen)
 {
    if (pw_addr && !pw_addrlen)
+   {
+      LOGC(cnlog.Error, log << "srt_accept: incorrect address parameters specified");
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1069]:
```suggestion
      LOGC(cnlog.Error, log << "srt_accept: provided address, but address length parameter is missing");
```

File: srtcore/api.cpp [hunk 3]
```diff
@@ -1065,16 +1065,25 @@ SRTSOCKET CUDTUnited::accept_bond(const SRTSOCKET listeners [], int lsize, int64
...
    if (ls == NULL)
+   {
+      LOGC(cnlog.Error, log << "srt_accept: invalid listener socket value: " << listen);
       throw CUDTException(MJ_NOTSUP, MN_SIDINVAL, 0);
+   }
 
    // the "listen" socket must be in LISTENING status
    if (ls->m_Status != SRTS_LISTENING)
+   {
+      LOGC(cnlog.Error, log << "srt_accept: socket @" << listen << " is not set listen state (forgot srt_listen?)");
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1084]:
```suggestion
      LOGC(cnlog.Error, log << "srt_accept: socket @" << listen << " is not in listening state (forgot srt_listen?)");
```

File: srtcore/api.cpp [hunk 4]
```diff
@@ -1123,15 +1132,22 @@ SRTSOCKET CUDTUnited::accept(const SRTSOCKET listen, sockaddr* pw_addr, int* pw_
    {
       // non-blocking receiving, no connection available
       if (!ls->m_pUDT->m_config.bSynRecving)
+      {
+         LOGC(cnlog.Error, log << "srt_accept: no pending connection available at the moment");
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1136]: This looks more like a warning. 🤔

File: srtcore/api.cpp [hunk 5]
```diff
@@ -1123,15 +1132,22 @@ SRTSOCKET CUDTUnited::accept(const SRTSOCKET listen, sockaddr* pw_addr, int* pw_
...
          throw CUDTException(MJ_AGAIN, MN_RDAVAIL, 0);
+      }
 
+      LOGC(cnlog.Error, log << "srt_accept: listener socket @" << listen << " is closed");
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1140]:
```suggestion
      LOGC(cnlog.Error, log << "srt_accept: listener socket @" << listen << " is already closed");
```

**Review by maxsharabayko — COMMENTED** [review]
Consider minor improvements to the perspicuity of logs.

File: srtcore/api.cpp (hunk 4 above)
Comment (ethouris) [reply srtcore/api.cpp:1136]: No. The `srt_accept` function in non-blocking mode should not even be called if the program didn't make sure first that the listener socket has a pending connection in wait.

**Pull request merged**